stdlib/public/core/String.swift

```diff
@@ -807,36 +807,41 @@ extension String {
 
     // TODO(String performance): Try out incremental case-conversion rather than
     // make UTF-16 array beforehand
+    enum DummyError: Error { case dummy }
     let codeUnits = Array(self.utf16).withUnsafeBufferPointer {
       (uChars: UnsafeBufferPointer<UInt16>) -> Array<UInt16> in
-      var result = Array<UInt16>(repeating: 0, count: uChars.count)
-      let len = result.withUnsafeMutableBufferPointer {
-        (output) -> Int in
+      var len: Int = 0
+      let result = try? Array<UInt16>(unsafeUninitializedCapacity: uChars.count) {
+        buf, initializedCount in
         var err = __swift_stdlib_U_ZERO_ERROR
-        return Int(truncatingIfNeeded:
-          __swift_stdlib_u_strToLower(
-            output.baseAddress._unsafelyUnwrappedUnchecked,
-            Int32(output.count),
-            uChars.baseAddress._unsafelyUnwrappedUnchecked,
-            Int32(uChars.count),
-            "",
-            &err))
+        len = Int(truncatingIfNeeded:
+        __swift_stdlib_u_strToLower(
+          buf.baseAddress._unsafelyUnwrappedUnchecked,
+          Int32(buf.count),
+          uChars.baseAddress._unsafelyUnwrappedUnchecked,
+          Int32(uChars.count),
+          "",
+          &err))
+        initializedCount = min(len, uChars.count)
+        if len > uChars.count {
+            throw DummyError.dummy
```

Comment:
Sorry, who catches this error? Does this code path represent a violation of an invariant or precondition?

Reply:
Rewritten without throwing.